sagemaker-python-sdk/keras_script_mode_pipe_mode_horovod/tensorflow_keras_CIFAR10.ipynb

```diff
@@ -0,0 +1,618 @@
+    "\n",
+    "in out script, we enabled Pipe Mode using the following code:\n",
+    "```python\n",
+    "from sagemaker_tensorflow import PipeModeDataset\n",
+    "dataset = PipeModeDataset(channel=channel_name, record_format='TFRecord')\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.tensorflow import TensorFlow\n",
+    "\n",
+    "\n",
+    "source_dir = os.path.join(os.getcwd(), 'source_dir')\n",
+    "estimator_pipe = TensorFlow(base_job_name='pipe-cifar10-tf',\n",
+    "                       entry_point='cifar10_keras_main.py',\n",
+    "                       source_dir=source_dir,\n",
+    "                       role=role,\n",
+    "                       framework_version='1.12.0',\n",
+    "                       py_version='py3',\n",
+    "                       hyperparameters=hyperparameters,\n",
+    "                       train_instance_count=1, train_instance_type='ml.p3.2xlarge',\n",
+    "                       tags = [{'Key' : 'Project', 'Value' : 'cifar10'},{'Key' : 'TensorBoard', 'Value' : 'pipe'}],\n",
+    "                       metric_definitions=keras_metric_definition,\n",
+    "                       input_mode='Pipe')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example, we'll configure ```wait=False``` if you want to see the output logs change to ```wait=True```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "remote_inputs = {'train' : inputs+'/train', 'validation' : inputs+'/validation', 'eval' : inputs+'/eval'}\n",
+    "estimator_pipe.fit(remote_inputs, wait=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Distributed training with horovod\n",
+    "Horovod is a distributed training framework based on MPI. Horovod is only available with TensorFlow version 1.12 or newer. You can find more details at Horovod README.\n",
+    "\n",
+    "To enable Horovod, we needed to add the following code to our script:\n",
+    "```python\n",
+    "import horovod.keras as hvd\n",
+    "hvd.init()\n",
+    "config = tf.ConfigProto()\n",
+    "config.gpu_options.allow_growth = True\n",
+    "config.gpu_options.visible_device_list = str(hvd.local_rank())\n",
+    "K.set_session(tf.Session(config=config))\n",
+    "```\n",
+    "\n",
+    "We added to following callbacks:\n",
+    "```python\n",
+    "hvd.callbacks.BroadcastGlobalVariablesCallback(0)\n",
+    "hvd.callbacks.MetricAverageCallback()\n",
+    "```\n",
+    "\n",
+    "Configured the optimizer:\n",
+    "```python\n",
+    "opt = Adam(lr=learning_rate * size, decay=weight_decay)\n",
+    "opt = hvd.DistributedOptimizer(opt)\n",
+    "```\n",
+    "and choose to save checkpoints and send TensorBoard logs only from the ```python hvd.rank() == 0``` instance\n",
```

Comment:
mixing of tenses: added, configured, choose. Suggest making all of these simple present tense: add, configure, choose